src/components/Login.tsx

```diff
@@ -2,43 +2,94 @@ import React, { useEffect, useState } from 'react'
 import { authInstance, firebaseAuth } from '../firebase/firebase'
 import { StyledFirebaseAuth } from 'react-firebaseui'
 import { Button } from 'react-bootstrap'
+import { connect } from 'react-redux'
+import * as actions from '../actions/actions'
+import { authEmulatorHost } from '../constants'
+import { fetchToken } from '../services/fetch-token'
+import { auth as firebaseuiAuth } from 'firebaseui'
 
 const uiConfig = {
     // Popup signin flow rather than redirect flow.
     signInFlow: 'popup',
     // Redirect to /signedIn after sign in is successful. Alternatively you can provide a callbacks.signInSuccess function.
     signInSuccessUrl: '/',
+    autoUpgradeAnonymousUsers: true,
     // We will display Google and Facebook as auth providers.
-    signInOptions: [firebaseAuth.GoogleAuthProvider.PROVIDER_ID],
+    signInOptions: [
+        firebaseAuth.GoogleAuthProvider.PROVIDER_ID,
+        firebaseAuth.EmailAuthProvider.PROVIDER_ID,
+        firebaseuiAuth.AnonymousAuthProvider.PROVIDER_ID,
+    ],
     callbacks: {
         // Avoid redirects after sign-in.
         signInSuccessWithAuthResult: () => false,
+        signInFailure: async function (error) {
+            console.log('Signin failure error: ', error)
+        },
     },
 }
 
-const Login = () => {
+const Login = ({ token, setToken }) => {
     const [isLoggedIn, setIsLoggedIn] = useState(false)
 
+    if (process.env.REACT_APP_LOCAL_DEVELOPMENT) {
```

Comment:
we could also have a general flags file that toggle those option and other if needed on the fly in one place.

Reply:
the variable is only used in local development environment, so I thought the appropriate place for it would be in the `.env.development` file. I did not use the `process.env.NODE_ENV` variable, since this way it is also possible to have a local development configuration that links to `dev` API

Reply:
lgtm, but we should really create a flags.js that store all runtime related logi, like api url and so on so we could switch easily, everything in one place.

Reply:
Is it because we don't currently have a specific local env file ?